src/App.tsx

```diff
@@ -15,48 +15,156 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [sortState, setSortState] = useState({
+    isUsable: false,
+    isAlphabetActive: false,
+    isLengthActive: false,
+    isReverseActive: false,
+  });
```

Comment:
Wouldn't it be easier to simply do:
```suggestion
  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
  const [isReverse, setIsReverse] = useState(false);
```

Reply:
I believe that by doing so, you will be able to 'untangle' your code because currently it's a mess

No offense ;) _"You are not your code"_